### PR TITLE
Support unconditional offers via the API

### PR DESCRIPTION
**Pull request by steventux**
## Context

We'd like to allow vendors to make unconditional offers via the API

## Changes proposed in this pull request

Branches out of the `AcceptOffer` service into an unconditional offers specific service so that we can avoid unnecessary Provider conditions checking states. (ie. transitions directly from `:offer` to `:recruited`) this is behind a feature flag.

## Guidance to review

Is the API specific feature flag name necessary? This could apply to any unconditional offer flow.

## Link to Trello card

https://trello.com/c/AuQprcmF/3422-support-unconditional-offers-via-the-api

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/accept_unconditional_offer.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    end
+
+    NotificationsList.for(@application_choice, include_ratifying_provider: true).each do |provider_user|
+      ProviderMailer.unconditional_offer_accepted(provider_user, @application_choice).deliver_later
+    end
+
+    CandidateMailer.unconditional_offer_accepted(@application_choice).deliver_later
+
+    StateChangeNotifier.new(:recruited, @application_choice).application_outcome_notification
+  end
```
Comment (cwrw) [inline app/services/accept_unconditional_offer.rb:19]: Maybe we can add a state change error rescue just incase: `rescue Workflow::NoTransitionAllowed`?

File: spec/services/accept_unconditional_offer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe AcceptUnconditionalOffer do
+  include CourseOptionHelpers
+
+  it 'sets the accepted_at date for the application_choice' do
```
Comment (cwrw) [inline spec/services/accept_unconditional_offer_spec.rb:6]: can we also add a test for `recruited_at` timestamp?

File: app/services/accept_unconditional_offer.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      @application_choice.update!(accepted_at: Time.zone.now, recruited_at: Time.zone.now)
+    end
+
+    NotificationsList.for(@application_choice, include_ratifying_provider: true).each do |provider_user|
+      ProviderMailer.unconditional_offer_accepted(provider_user, @application_choice).deliver_later
+    end
+
+    CandidateMailer.unconditional_offer_accepted(@application_choice).deliver_later
+
+    StateChangeNotifier.new(:recruited, @application_choice).application_outcome_notification
```
Comment (cwrw) [inline app/services/accept_unconditional_offer.rb:18]: Not sure but is the state change notifier something we usually/can test for?

Comment (steventux) [reply app/services/accept_unconditional_offer.rb:18]: :+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4154/commits/45abd80da804b4b7ba26030e77c8686870e86214

File: spec/services/accept_unconditional_offer_spec.rb (hunk 2 above)
Comment (steventux) [reply spec/services/accept_unconditional_offer_spec.rb]: :+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4154/commits/076ccbcee103bd0dd06383c5f04761f62314d8f6

File: app/services/accept_unconditional_offer.rb (hunk 1 above)
Comment (steventux) [reply app/services/accept_unconditional_offer.rb:21]: :+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4154/commits/d380acc8838ead576966a77f7df357d1e6a8f7c2

**Review by cwrw — APPROVED** [review]
Looks good, thanks for addressing comments 👍 
Only comment is around adding mail previews but we can add those in subsequent prs

**Pull request merged**